lib/stages/basebandReadout.cpp

```diff
@@ -141,19 +145,33 @@ void basebandReadout::main_thread() {
             auto first_meta = (chimeMetadata*)buf->metadata[buf_frame]->metadata;
 
             stream_id_t stream_id = extract_stream_id(first_meta->stream_ID);
-            uint32_t freq_id = bin_number_chime(&stream_id);
-
-            DEBUG("Initialize baseband metrics for freq_id: {:d}", freq_id);
-            readout_counter.labels({std::to_string(freq_id), "done"});
-            readout_counter.labels({std::to_string(freq_id), "error"});
-            readout_counter.labels({std::to_string(freq_id), "no_data"});
-            readout_in_progress_metric.labels({std::to_string(freq_id)}).set(0);
-
-            INFO("Starting request-listening thread for freq_id: {:d}", freq_id);
-            mgr = &basebandApiManager::instance().register_readout_stage(freq_id);
-            lt = std::make_unique<std::thread>([&] { this->readout_thread(freq_id, *mgr); });
-
-            wt = std::make_unique<std::thread>([&] { this->writeout_thread(*mgr); });
+            INFO("slot_id:{:d}",stream_id.slot_id);            
+            INFO("link_id:{:d}",stream_id.link_id);            
+            uint32_t freq_id=0;
+		for(int freqidx=0; freqidx < _num_local_freq;freqidx++){
+		    // XXX Map stream IDs to freq IDs with bin_number() (for Pathfinder) or bin_number_chime()
+		    // XXX Use num_local_freqs to figure out if we're running on CHIME or PF
+		    if (_num_local_freq == 1){
+			freq_id = bin_number_chime(&stream_id);
+		    } else { // more than one freq per stream
```

Comment:
`bin_number` only works for the pathfinder configuration right? I think there are 3 valid cases: chime, pathfinder, and 16 input.

Reply:
Yes, I am implementing this in fpga_header_functions.c as `bin_number_multifreq(num_local_freq, stream_id, freqidx)`.